packages/block-library/src/block/use-no-recursive-renders.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	createContext,
+	useCallback,
+	useContext,
+	useMemo,
+} from '@wordpress/element';
+
+const RenderedRefsContext = createContext( [] );
+
+export default function useNoRecursiveRenders( ref ) {
+	const previouslyRenderedRefs = useContext( RenderedRefsContext );
```

Comment:
Not a blocker for this PR. I wanted to share my thought on how to make it general enough to use with other blocks that require the same guard (Post Content or Template Part). It looks like it would be enough to rename `ref` to something more generic like `uniqueId` and use `Set` rather than array of strings for checks to support compound values for more complex blocks.

Reply:
Agreed, thanks for the note!